Drv/LinuxSpiDriver/LinuxSpiDriverComponentImpl.cpp

```diff
@@ -97,9 +98,9 @@
 
         // Configure:
         /*
-         * SPI Mode 0
+         * SPI Mode 0, 1, 2, 3
          */
-        U8 mode = SPI_MODE_0; // Mode 0 (CPOL = 0, CPHA = 0)
+        U8 mode = spiMode; // Mode Select (CPOL = 0/1, CPHA = 0/1)
```

Comment:
For small enums, it's better IMHO to use a switch statement to cover the values (that pass) and a default clause with the assert. Thoughts @LeStarch?

Reply:
Switch to a switch statement e.g. :
```
switch (spiMode) {
    case SpiMode::SPI_MODE_0:
         mode = SPI_MODE_0;
         break;
...
    default:
        FW_ASSERT(...)
       break;
}
ret = ioctl(fd, SPI_IOC_WR_MODE, &mode);

```

Does get around the hard-coded values in the enum definition.  It would also protect us from changes to the kernel #defs, although I don't see how that would happen.

Reply:
I am ok with making it a switch statement.

Reply:
@timcanham @LeStarch I pushed a commit which utilizes a switch statement.